IS0402Test.py

```diff
@@ -1115,7 +1115,122 @@ def test_22_1(self, test):
+            else:
+                sub_json = self.prepare_subscription("/nodes")  # No downgrade for version under test
+            resp_json = self.post_subscription(test, sub_json)
+            websockets[api_version] = WebsocketWorker(resp_json["ws_href"])
+            websockets[api_version].start()
+
+        sleep(WS_MESSAGE_TIMEOUT)  # Wait for SYNC messages
+
+        # Verify no error occurred on starting websocket subscription & clear SYNC messages
+        for api_version in query_versions:
+            if websockets[api_version].did_error_occur():
+                return test.FAIL("Error opening websocket: {}".format(websockets[api_version].get_error_message()))
+            websockets[api_version].clear_messages()
+
+        # Register a Node at each API version available (up to the version under test)
+        node_ids = {}
+        test_id = str(uuid.uuid4())
+        for api_version in query_versions:
+            # Note: We iterate over the Query API versions, not the Reg API as it's the Query API that's under test
+            test_data = self.copy_resource("node", api_ver=api_version)
+            test_data["id"] = str(uuid.uuid4())
+            test_data["description"] = test_id
+            node_ids[api_version] = test_data["id"]
+
+            reg_url = "{}/{}/".format(self.reg_url.rstrip(reg_api["version"] + "/"), api_version)
+            self.post_resource(test, "node", test_data, codes=[201], reg_url=reg_url)
+
+        sleep(WS_MESSAGE_TIMEOUT)
+
+        # Read data & close websockets
+        sub_data = dict()
+        for api_version in query_versions:
+            received_messages = websockets[api_version].get_messages()
+            websockets[api_version].close()
+            grain_data = list()
+
+            for curr_msg in received_messages:
+                json_msg = json.loads(curr_msg)
+                grain_data.extend(json_msg["grain"]["data"])
+
+            sub_data[api_version] = grain_data
+
+        # Verify that all POSTed Nodes are announced via the corresponding websocket sub as expected (incl downgrade)
+        for api_version in query_versions:
+            expected_nodes = []
+            for node_api_version, node_id in node_ids.items():
+                if self.is04_query_utils.compare_api_version(node_api_version, api_version) >= 0:
+                    expected_nodes.append(node_id)
+            for expected_node_id in expected_nodes:
+                found_data_set = False
+                for curr_data in sub_data[api_version]:
+                    if "pre" not in curr_data and "post" in curr_data:  # Only check the 'Added Event'
+                        if "id" in curr_data["post"]:
+                            if expected_node_id == curr_data["post"]["id"]:
+                                found_data_set = True
+                                break
+
+                if not found_data_set:
```

Comment:
Just wondering if it would be worth adding an extra check here to make sure we don't see any Nodes which we wouldn't expect to? Otherwise is there a danger that all Nodes may be exposed via every WebSocket under test?

Reply:
Makes sense -- is on my todo

Reply:
-> Follow up: #384